config/remote_config.go

```diff
@@ -0,0 +1,189 @@
+package config
+
+import (
+	"bufio"
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"math/big"
+	m "math/rand"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type RemoteConfig struct{}
+
+type FileRemovalStatus int
+
+const (
+	FileNotFound FileRemovalStatus = iota
+	FileRemoved
+	FileIsDirectory
+	PermissionDenied
+	UnknownError
+)
+
+func NewRemoteConfig() *RemoteConfig {
+	return &RemoteConfig{}
+}
+
+func (c *RemoteConfig) WriteTo(rawURL, targetPath string) error {
+	var (
+		resp []byte
+		err  error
+	)
+
+	resp, err = retry(context.Background(), 3, func(ctx context.Context) ([]byte, error) {
```

Comment:
```suggestion
func (c *RemoteConfig) WriteTo(ctx context.Context, rawURL, targetPath string) error {
	var (
		resp []byte
		err  error
	)

	resp, err = retry(ctx, 3, func(ctx context.Context) ([]byte, error) {
```

Lets the caller define the timeout and such :ok_hand: